picard/formats/id3.py

```diff
@@ -650,36 +650,30 @@ def _save_tags(self, tags, filename):
             tags.update_to_v24()
             tags.save(filename, v2_version=4, v1=v1)
 
-    @property
-    def new_metadata(self):
-        config = get_config()
-        if not config.setting["write_id3v23"]:
-            return self.metadata
-
-        copy = Metadata()
-        copy.copy(self.metadata)
+    def format_specific_metadata(self, metadata, tag, settings=None):
+        if not settings:
+            settings = get_config().setting
```

Comment:
Is there any material difference between `get_config().setting.as_dict()` which returns a dict and `get_config().setting` which returns a ConfigSection object? Should we return a dict here too?

Reply:
The difference is in performance, `as_dict` is rather expensive. This is not good:

```python
settings = get_config().setting.as_dict()
if settings['foo']:
   do_something()
```

The config value `foo`  is accessed only once, `as_dict` has to read and copy all values from dict. It makes sense if you have something like this:

```python
settings = get_config().setting.as_dict()
for i in range(0, 10000):
    if settings['foo']:
        do_something()
```

Now the costs of accessing `setting[]` would be way greater than the costs of the dict conversion.

Of course the following would be even better:

```python
settings = get_config().setting
foo = settings['foo']
for i in range(0, 10000):
    if foo:
        do_something()
```

But in this case we don't do this, because we don't know in advance what settings are needed in the loop. It's a compromise, but as_dict is fast enough that it doesn't matter much in the case where it is called only once.